language_detector/main.py

```diff
@@ -1,9 +1,27 @@
+# -*- coding: utf-8 -*-
+
 """This is the entry point of the program."""
 
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
-    """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    """Returns the detected lang of given text."""
+    numOfTimesLanguageOccurs = {}
+
+    for word in text.split():
+        lang = whichLanguageIsIt(word, languages)
+        if lang is not None:
+            addToDictionary(lang, numOfTimesLanguageOccurs)
+    return getHighestOccurringLanguage(numOfTimesLanguageOccurs)
+
+def whichLanguageIsIt(word, language_list ):
+    for dict in language_list:
+        if word in dict['common_words']: 
+            return dict['name']
+
+def addToDictionary(lang, dictionary):
+        if lang in dictionary: dictionary[lang] += 1
+        else: dictionary[lang] = 1
+
```

Comment:
dictionary.get() is a great replacement for this kind of function. [dict.get()](http://www.tutorialspoint.com/python/dictionary_get.htm). For example: 

    dictionary [lang] = dictionary.get(lang, 0)  + 1

This code asks get to return the value of the key lang and add 1 to that value. If lang does not exist it will create it and give it a value of 1.